src/program.js

```diff
@@ -20,10 +21,15 @@ export class Program {
   yargs: any;
   commands: { [key: string]: Function };
   shouldExitProgram: boolean;
+  checkForUpdates: typeof checkForAutomaticUpdates;
 
   constructor(
     argv: ?Array<string>,
-    {absolutePackageDir = process.cwd()}: {absolutePackageDir?: string} = {}
+    {
+      absolutePackageDir = process.cwd(),
```

Comment:
similarly, this module / class should be set up for dependency injection so that you can replace the update utility with a stub for testing:
````javascript
import {checkForAutomaticUpdates as defaultUpdateChecker} from './util/updates';

class Program {
  constructor(
    argv: ?Array<string>,
    {
      absolutePackageDir?: string,
      checkForAutomaticUpdates = defaultUpdateChecker,
    }
  ) {
    // ...
    this.checkForUpdates = checkForAutomaticUpdates;
  }
}
````

The test can then use a stub:
````javascript
const checkForAutomaticUpdates = sinon.stub();
const program = new Program(['run'], {checkForAutomaticUpdates});
await program.execute();

assert.equal(checkForAutomaticUpdates.firstCall.args[0].version, '1.0.0');
````